nordicsemi/dfu/dfu_transport_ble.py

```diff
@@ -616,15 +625,28 @@ def try_to_recover():
                 break
             else:
                 raise NordicSemiException("Failed to send firmware")
-            self._send_event(event_type=DfuEvent.PROGRESS_EVENT, progress=len(data))
+            self.current_dfu_stage.progress = min(((current_offset + response['max_size']) / len(firmware)) * 100, 100)
+            logger.debug(f"Current progress: {self.current_dfu_stage.progress:.2f} %")
```

Comment:
What do you think about handling the `progress` with setter? It could take care about e.g. the 100% limit (`min(calculated, 100)`) and logging the set value.